components/org.wso2.carbon.identity.application.authenticator.fido2/src/main/java/org/wso2/carbon/identity/application/authenticator/fido2/core/WebAuthnService.java

```diff
@@ -928,8 +928,10 @@ private UserStoreManager getUserStoreManager(User user) throws UserStoreExceptio
 
     private UserIdentity buildUserIdentity(User user) throws FIDO2AuthenticatorServerException {
 
+        ByteArray userHandle = FIDO2DeviceStoreDAO.getInstance().getUserHandleForUsername(user.toString())
```

Comment:
Hope this doesn't have an effect on already generated userHandles. Need to make sure the login flow works properly for already registered users after this change.

Reply:
Yes this won't have an effect on the already registered users, as we are only changing the registration flow and this will not effect the authentication process. However in the master branch if the users have configured multiple FIDO devices, they are only able to authenticate using only one device(this issue has been introduced after the major version upgrade done in Yubico webAuthn). However already registered users(with multiple devices) who are having trouble with login with multiple devices, that issue won't be resolve unless they reconfigure devices.